my_enumerable.rb

```diff
@@ -0,0 +1,91 @@
+module Enumerable
+  def my_each
+    for item in self do
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    for item in self do
+      new_arr.push(item) if yield(item)
+    end
+    new_arr
+  end
+
+  def my_all?
+    result = true
+    for item in self do
+      return false if yield(item) == false
+    end
+    result
+  end
+
+  def my_any?
+    for item in self do
+      return true if yield(item)
+    end
+    false
+  end
+
+  def my_none?
+    for item in self do
+      return false if yield(item)
+    end
+    true
+  end
+
+  def my_count
+    counter = 0
+    for item in self
+      counter += 1 if yield(item)
+    end
+    counter
+  end
+
+  def my_map
+    new_arr = []
+    for item in self do
+      new_arr << yield(item)
+    end
+    new_arr
+  end
+
+  def my_map_with_proc
+    new_arr = []
+    for item in self do
+      new_arr << proc.call(item)
+    end
+    new_arr
+  end
+
+  def my_map_with_proc_or_block
+    new_arr = []
+    for item in self
+      new_arr << proc.call(item) || yield(item)
+    end
+    new_arr
+  end
+
+  def my_inject
+    sum = self[0]
+    for item in self do
+      sum = yield(sum, item)
+    end
+    sum
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
array = Array.new(100) { rand(0...9) }
array.my_inject(:+) == array.inject(:+) # should return true
```
- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
operation = proc { |sum, n| sum + n } 
search = proc { |memo, word| memo.length > word.length ? memo : word } 
array = Array.new(100){rand(0...9)}
words = %w[dog door rod blade]

array.my_inject(&operation) == array.inject(&operation) # should return true
actual = range.my_inject(4) { |prod, n| prod * n }
expected = range.inject(4) { |prod, n| prod * n }
actual == expected #true
words.my_inject(&search) == words.inject(&search) # should return true
```